File: Leetcode/Week1/Array/C++/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        vector<int> vec;
        
        for(int i =0; i <nums.size(); i++) {
            if(mp.find(target - nums[i]) != mp.end()){
                vec.push_back(mp[target - nums[i]]);
                vc.push_back(i);
            } else {
                mp[nums[i]] = i;
            }
        }
        return vec;
    }
};

File: Leetcode/Week1/Array/Java/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> elements = new HashMap<Integer, Integer>();
        int[] res = new int[2];
        for(int i = 0; i < nums.length; i++) {
            if(elements.containsKey(target - nums[i])) {
                res[0] = elements.get(target - nums[i]);
                res[1] = i;
            } else {
                elements.put(nums[i], i);
            }
        }
        return res;
    }
}